### DISCUSSION TITLE
Monitoring

### DISCUSSION THREAD

**Discussion by PureMSP** [postalserver/postal#1473] [Feature suggestions]
Hi.
Firstly, this software is just great! Works a treat.
Ok. I've implemented this for a client of mine. All is working as expected.. 

We use Zabbix for some of our monitoring, and data collection.. I was wondering if anyone can assist in the best method for reporting on statistics, like emails out, bounce count, queue length etc etc etc. Normally I'd use a postfix template to gather all the numbers..

Thanks in advance.

**Comment by willpower232** [comment]
Unfortunately such information isn't available outside of the UI yet (https://github.com/atech/postal/issues/453).

Monitoring the underlying server can help seeing issues in the meantime, we've been able to spot problems when there is unexpected increased MySQL activity using Zabbix and Grafana for example.

**Comment by PureMSP** [comment]
Ah. Ok. MySQL is ok. I can monitor that..

I'm guessing the the application itself is bespoke, rather than porting off the likes of Postfix, or an equivalent?

**Comment by PureMSP** [comment]
Also, what information does the MySQL DB hold?

**Comment by willpower232** [comment]
The application is bespoke indeed, all the application information is kept in MySQL. Each mailserver has its own database with the main database remembering which database is which.

**Comment by arthurzenika** [comment]
We have a small bash script that extracts information from the postal logs to track how many email are going through, not great but the hack works for us : 

```
cat /etc/zabbix/scripts/postal_stats.sh 
#!/bin/bash

DATE=$(date '+%Y-%m-%d')

cat /var/log/postal/smtp/smtp_server.log* |grep -a "MAIL FROM:" |grep -c "$DATE"
```
Hope this can help others.

**Reply by ariveamar** [reply]
It was great thank you
